runbotics/runbotics-scheduler/src/scheduler-database/attribute/attribute.service.ts

```
@@ -0,0 +1,124 @@
+import { SecretService } from '../secret/secret.service';
+
+const relations = ['tenant', 'createdBy', 'updatedBy'];
+
+@Injectable()
+export class AttributeService {
+  private readonly logger = new Logger(AttributeService.name);
+
+  constructor(
+    @InjectRepository(Attribute)
+    private readonly attributeRepository: Repository<Attribute>,
+    private readonly secretService: SecretService,
+  ) { }
+
+  async create(attributeDto: CreateAttributeDto, request: AuthRequest): Promise<Attribute> {
+    const { user: { tenantId, id: userId } } = request;
+
+    const encryptedValue = this.secretService.encrypt(attributeDto.value, tenantId);
+
+    const secretEntity = {
+      ...encryptedValue,
+      value: encryptedValue,
+      tenantId,
+      createdById: userId,
+      updatedById: userId,
+    };
+
+    const secret = await this.secretService.save(secretEntity);
+      // .catch((error) => {
+      //   this.logger.error(error);
+      //   throw new BadRequestException(`Failed to save secret: ${error.message}`);
+      // });
+    const secretId = secret.id;
+
+    const attribute = {
+      ...attributeDto,
+      secretId: secretId,
+      tenantId,
+      createdById: userId,
+      updatedById: userId,
+      masked: attributeDto.masked || true,
+    };
+
+    return this.attributeRepository.manager.save(Attribute, attribute)
+      .catch((error) => {
+        this.logger.error(error);
+        throw new BadRequestException(`Failed to save attribute: ${error.message}`);
+      });
+  }
+
+  async findAll(): Promise<Attribute[]> {
+    const attributes = await this.attributeRepository.find({
+      relations,
+    });
+
+    return attributes;
+  }
+
+  async findOneById(id: string): Promise<Attribute> {
+    const attribute = this.attributeRepository.findOne({
+      where: {
+        id,
+      },
+      relations,
+    });
+
+    if (!attribute) {
+      throw new NotFoundException(`Attribute with id: ${id} not found`);
+    }
```

I am not suer if you always want to throw exception if attribute is not found, I wander if this case shouldn't be handled outside of the AttributeService